test/unit/permutations/test_permutations.cpp

```diff
@@ -0,0 +1,113 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include "gtest/gtest.h"
+
+#include "dlaf/matrix/matrix_mirror.h"
+#include "dlaf/permutations/general.h"
+#include "dlaf/permutations/general/impl.h"
+#include "dlaf/util_matrix.h"
+
+#include "dlaf_test/matrix/util_matrix.h"
+#include "dlaf_test/matrix/util_tile.h"
+#include "dlaf_test/util_types.h"
+
+using namespace dlaf;
+using namespace dlaf::test;
+
+template <typename Type>
+class PermutationsTestCPU : public ::testing::Test {};
+TYPED_TEST_SUITE(PermutationsTestCPU, MatrixElementTypes);
+
+#ifdef DLAF_WITH_CUDA
+template <typename Type>
+class PermutationsTestGPU : public ::testing::Test {};
+TYPED_TEST_SUITE(PermutationsTestGPU, MatrixElementTypes);
+#endif
+
+// The portion of the input matrices rows/columns specified by `i_begin` and `i_end` are placed in
+// reverse order into the output matrix.
+template <Backend B, Device D, class T, Coord C>
+void testPermutations(SizeType n, SizeType nb, SizeType i_begin, SizeType i_end) {
+  Matrix<SizeType, Device::CPU> perms(LocalElementSize(n, 1), TileElementSize(nb, 1));
+  Matrix<T, Device::CPU> mat_in_h(LocalElementSize(n, n), TileElementSize(nb, nb));
+  Matrix<T, Device::CPU> mat_out_h(LocalElementSize(n, n), TileElementSize(nb, nb));
+
+  const matrix::Distribution& distr = mat_out_h.distribution();
+
+  SizeType index_start = distr.globalElementFromGlobalTileAndTileElement<C>(i_begin, 0);
+  SizeType index_finish = distr.globalElementFromGlobalTileAndTileElement<C>(i_end, 0) +
+                          distr.tileSize(GlobalTileIndex(i_end, i_end)).get<C>();
+  dlaf::matrix::util::set(perms, [index_start, index_finish](GlobalElementIndex i) {
+    if (index_start > i.row() || i.row() >= index_finish)
+      return SizeType(0);
+
+    return index_finish - 1 - i.row();
+  });
+  dlaf::matrix::util::set(mat_in_h, [](GlobalElementIndex i) {
+    return T(i.get<C>() - i.get<orthogonal(C)>()) / T(8);
```

Comment:
```suggestion
    return T(i.get<C>()) - T(i.get<orthogonal(C)>()) / T(8);
```
I suggested `c - o / 8` such that to get the same sequence of elements in two nearest rows/cols you should move by 8 cols/rows. Using `(c - o) / 8` increasing the rows and cols by the same amount gives the same sequence.

Reply:
https://github.com/eth-cscs/DLA-Future/pull/564/commits/f41ebd0f32551a8e0d6726d94b0345f45cf7a99e